### PR TITLE
Add support for Google Cloud Datastore

### PR DESCRIPTION
**Pull request by pingvinen**
This adds a new package called `RapidCore.GoogleCloud`. This currently only contains stuff for working with [Google Cloud Datastore](https://cloud.google.com/datastore).

It contains both an ORM and a connection class (`DatastoreConnection`) which is following the concept of the `MongoDbConnection`.

The ORM can be used without using DatastoreConnection!

The current implementation is first and foremost to get it working. It has not been optimized for performance in any way.

### CODE REVIEW

**Comment by pingvinen** [comment]
@nover I have rebased with master and added documentation + some tweaks for things I noticed while writing the documentation. Will you take a look?

File: docs/Datastore/Connection.md [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+```csharp
+public class KewlPoco
+{
+    public string Id { get; set; }
+    public string Oh { get; set; }
+    public int IntOfDoom { get; set; }
+}
+
+// load the document
+var poco = ...load entity...;
```
Comment (nover) [inline docs/Datastore/Connection.md:47]: I think we should provide a real example here on how to load the code. We must remember that the users coming in to the documentation are probably new to the ORM..

**Review by nover — CHANGES_REQUESTED** [review]
Overall I think it looks good. I have to admit that I have only looked at this from the compiled documentation point of view though... Based on this I have a few questions / comments though:

1. The new package is missing from the index on the documentation page
1. How does one set the `namespace` the Datastore connection should be operating within? 
1. Should the docs explicitly state that there is recursion protection so that circular object graphs will throw an exception? Slash, saying that circular object graphs will _not_ persist properly and the user has to handle it and make the graph non-circular.

File: docs/Datastore/Connection.md (hunk 1 above)
Comment (pingvinen) [reply docs/Datastore/Connection.md:47]: As I recall the next example on that page is actually a "load an entity"-example.

**Comment by pingvinen** [comment]
> How does one set the namespace the Datastore connection should be operating within

That is shown in the "Creating an instance"-example on the connection page.
```csharp
using RapidCore.GoogleCloud.Datastore;

var datastoreDb = Google.Cloud.Datastore.V1.DatastoreDb.Create("project-id", "namespace");
var connection = new DatastoreConnection(datastoreDb);
```

> Should the docs explicitly state that there is recursion protection so that circular object graphs will throw an exception? Slash, saying that circular object graphs will not persist properly and the user has to handle it and make the graph non-circular.

I am not sure. On the one hand I as a consumer expect that there is protection, and I find it to be a weird request to be able to save an infinite recursion.

**Review by nover — APPROVED** [review]

**Pull request merged**